src/coreComponents/physicsSolvers/wavePropagation/pygeosx/functions.py

```diff
@@ -0,0 +1,221 @@
+'''
+Created on 8/02/2021
+
+@author: macpro
+'''
+
+import math as m
+from functions import *
+import numpy as np
+from mesh import *
+from acquisition import *
+import pygeosx
+import sys
+from mpi4py import MPI
+from matplotlib import pyplot as plt
+from segy import *
+from print import *
+
+    
+    
+def callback(matrix, array):
```

Comment:
Sorry in case it wasn't clear, you only need to register a callback if you're using an implicit solver and want access to the matrix before the solve.

Reply:
Ok I was actually not sure of what it was doing, thanks for the clarification.